src/js/first-run/first-run.js

```diff
@@ -1,7 +1,35 @@
 "use strict";
 
+function generatePKCECodeVerifierValue() {
+  /*
+   * A code verifier is a random string using characters of
+   * [A-Z] / [a-z] / [0-9] / "-" / "." / "_" / "~",
+   * with a minimum length of 43 characters and a maximum length of 128 characters.
+   */
+  var text = "";
+  var possible = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
+
+  for (var i = 0; i < Math.random() * (128 - 43) + 43; i++) {
+    text += possible.charAt(Math.floor(Math.random() * possible.length));
+  }
+
+  return text;
+}
```

Comment:
Does "random" here mean "cryptographically secure random"? Because if so, `Math.random` doesn't provide that. (I'm a bit nervous about not using a library for authn algorithms anyway...)

And a nit to avoid var and use consistent capitalisation:

```suggestion
function generatePkceCodeVerifierValue() {
  /*
   * A code verifier is a random string using characters of
   * [A-Z] / [a-z] / [0-9] / "-" / "." / "_" / "~",
   * with a minimum length of 43 characters and a maximum length of 128 characters.
   */
  let text = "";
  const possible = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";

  for (let i = 0; i < Math.random() * (128 - 43) + 43; i++) {
    text += possible.charAt(Math.floor(Math.random() * possible.length));
  }

  return text;
}
```

Reply:
How does https://www.npmjs.com/package/pkce-challenge look?

Reply:
I don't have experience with any, unfortunately. pkce-challenge looks like it's made for Node.js, so probably won't work in the add-on. Checking with a former colleague, PKCE is limited enough that we might be able to get away with implementing it ourselves, so we could maybe just use [`window.crypto.getRandomValues`](https://developer.mozilla.org/docs/Web/API/Crypto/getRandomValues).

Reply:
RE: Bundler question from a month ago: There's an extra submission/review step if we use a bundler. That may not be as heavy as a lift, but every add-on I've worked on at Mozilla, we didn't have one. 

https://extensionworkshop.com/documentation/publish/source-code-submission/

Reply:
TIL about `window.crypto.getRandomValues`!